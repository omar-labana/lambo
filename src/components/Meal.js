import pizza from '../assets/pizza.png'

const Meal = () => {
    return (
        <div className="h-96 w-full bg-white flex">
            <img src='https:\/\/www.themealdb.com\/images\/media\/meals\/020z181619788503.jpg' className="w-1/3"></img>
            <div className="w-2/3 flex flex-col h-full justify-between p-10">
                <h2 className="text-5xl dancing-font font-bold">Discover</h2>
                <h3 className="text-5xl font-semibold ">Meal Name</h3>
                <p className="">
                    The chicken is a type of domesticated fowl, a subspecies of the red junglefowl (Gallus gallus). It is one of the most common and widespread domestic animals, with a total population of more than 19 billion as of 2011. There are more chickens in the world than any other bird or domesticated fowl.
                </p>


                <a className="text-right yellow text-xl font-bold">More Details &#8594;</a>
            </div>
        </div>
    )
}

export default Meal
