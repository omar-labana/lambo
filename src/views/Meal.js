import React, { useEffect } from 'react';
import { useLocation } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { getDish } from '../redux/slices/dish';

const Meal = () => {
  const { state } = useLocation();
  const { id } = state;

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getDish({ id }));
  }, [dispatch]);

  const dish = useSelector((state) => state.dish.dish);
  const {
    strArea, strCategory, strInstructions, strMeal, strMealThumb,
  } = dish;
  const {
    strIngredient1, strIngredient2, strIngredient3, strIngredient4, strIngredient5,
  } = dish;

  return (
    <div className="text-white text-2xl flex flex-col items center gap-10 mx-20">
      <p className="dancing-font text-7xl">{strMeal}</p>
      <div className="flex justify-around">
        <div className="text-3xl flex flex-col gap-10">
          <p>
            Country:
            <span className="dancing-font">{strArea}</span>
          </p>
          <p>
            Category:
            <span className="dancing-font">{strCategory}</span>
          </p>
        </div>
        <img className="w-1/4" src={strMealThumb} alt={strMeal} />
      </div>
      <div className="">
        <p><span className="dancing-font text-4xl">strIngredients: </span></p>
        <ul>
          <li>{strIngredient1}</li>
          <li>{strIngredient2}</li>
          <li>{strIngredient3}</li>
          <li>{strIngredient4}</li>
          <li>{strIngredient5}</li>
        </ul>
      </div>
      <p>
        <span className="dancing-font text-4xl">How to preapare: </span>
        {strInstructions}
      </p>
    </div>
  );
};

export default Meal;
