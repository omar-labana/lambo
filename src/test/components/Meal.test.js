import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Meal from '../../components/Meal';

const stl = {
  strMeal: '1',
  strMealThumb: '1',
  type: '1',
  idMeal: '1',
};

describe('Meal Component', () => {
  const wrapper = shallow(<Meal meal={stl} key={stl.idMeal} />);

  it('exists', () => {
    expect(wrapper.find('Meal')).toBeTruthy();
  });

  it('has correct props', () => {
    const propa = wrapper.props();
    expect(propa).toBeTruthy();
  });

  it('has to match snap', () => {
    const button = renderer.create(
      <BrowserRouter><Meal meal={stl} key={stl.idMeal} /></BrowserRouter>,

    );
    expect(button).toMatchSnapshot();
  });
});
