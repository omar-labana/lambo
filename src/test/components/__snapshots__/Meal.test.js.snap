// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meal Component has to match snap 1`] = `
<div
  className="h-96 w-full bg-white flex"
>
  <img
    alt="1"
    className="w-1/3 object-cover"
    src="1"
  />
  <div
    className="w-2/3 flex flex-col h-full justify-between p-10"
  >
    <h2
      className="text-5xl dancing-font font-bold"
    >
      1
    </h2>
    <h3
      className="text-5xl font-semibold "
    >
      1
    </h3>
    <a
      className="text-right yellow text-xl font-bold"
      href="/meal/1"
      onClick={[Function]}
    >
      More Details â†’
    </a>
  </div>
</div>
`;
